packages/api/test/__fixtures__/sdk/petstore/index.ts

```diff
@@ -0,0 +1,617 @@
+   *
+   * @summary Find purchase order by ID
+   */
+  get(path: string, metadata: GetOrderByIdMetadataParam): Promise<Order>;
+  /**
+   * Logs user into the system
+   *
+   */
+  get(path: string, metadata: LoginUserMetadataParam): Promise<LoginUser_Response_200>;
+  /**
+   * Get user by user name
+   *
+   */
+  get(path: string, metadata: GetUserByNameMetadataParam): Promise<User>;
+  /**
+   * Access any get endpoint on your API.
+   *
+   * @param path API path to make a request against.
+   * @param metadata Object containing all path, query, header, and cookie parameters to supply.
+   */
+  get<T = unknown>(path: string, metadata?: Record<string, unknown>): Promise<T> {
+    return this.core.fetch(path, 'get', metadata);
+  }
+
+  /**
+   * Deletes a pet
+   *
+   */
+  delete<T = unknown>(path: string, metadata: DeletePetMetadataParam): Promise<T>;
+  /**
+   * For valid response try integer IDs with positive integer value. Negative or non-integer values will generate API errors
+   *
+   * @summary Delete purchase order by ID
+   */
+  delete<T = unknown>(path: string, metadata: DeleteOrderMetadataParam): Promise<T>;
+  /**
+   * This can only be done by the logged in user.
+   *
+   * @summary Delete user
+   */
+  delete<T = unknown>(path: string, metadata: DeleteUserMetadataParam): Promise<T>;
+  /**
+   * Access any delete endpoint on your API.
+   *
+   * @param path API path to make a request against.
+   * @param body Request body payload data.
+   * @param metadata Object containing all path, query, header, and cookie parameters to supply.
+   */
+  delete<T = unknown>(path: string, body?: unknown, metadata?: Record<string, unknown>): Promise<T> {
+    return this.core.fetch(path, 'delete', body, metadata);
+  }
+
+  /**
+   * Logs out current logged in user session
+   *
+   */
+  logoutUser<T = unknown>(): Promise<T> {
+    return this.core.fetch('/user/logout', 'get');
+  }
+
+  /**
+   * Returns a map of status codes to quantities
+   *
+   * @summary Returns pet inventories by status
+   */
+  getInventory(): Promise<GetInventory_Response_200> {
+    return this.core.fetch('/store/inventory', 'get');
+  }
+
+  /**
+   * Add a new pet to the store
+   *
+   */
+  addPet<T = unknown>(body: Pet): Promise<T> {
+    return this.core.fetch('/pet', 'post', body);
+  }
+
+  /**
+   * Update an existing pet
+   *
+   */
+  updatePet<T = unknown>(body: Pet): Promise<T> {
+    return this.core.fetch('/pet', 'put', body);
+  }
+
+  /**
+   * Updates a pet in the store with form data
+   *
+   */
+  updatePetWithForm<T = unknown>(
+    body: UpdatePetWithFormFormDataParam,
+    metadata: UpdatePetWithFormMetadataParam
+  ): Promise<T>;
+  /**
+   * Updates a pet in the store with form data
+   *
+   */
+  updatePetWithForm<T = unknown>(metadata: UpdatePetWithFormMetadataParam): Promise<T>;
+  /**
+   * Updates a pet in the store with form data
+   *
+   */
+  updatePetWithForm<T = unknown>(
```

Comment:
I really do not like these duplicate docblocks but I also don't really know how better to handle it. Maybe the docblock could go on the core/non-overloaded method only?

Reply:
at least it's codegen so they will be in sync. It would be nice if we could dedupe though with `@see` or something similar.